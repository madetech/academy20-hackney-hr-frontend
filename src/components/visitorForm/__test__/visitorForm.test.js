import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { screen } from '@testing-library/react';
import VisitorForm from '../visitorForm';
import renderer from 'react-test-renderer';

let container = null;
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null
});

describe("visitorForm", () => {
    test('renders the visitor form without crashing', () => {
      render(<VisitorForm />, container);
    });
    test('renders the form', () => {
        render(<VisitorForm />, container);
        const formElement = screen.getByText(/First name:/i);
        expect(formElement).toBeInTheDocument();
      });
})

test('Visitor form is rendered', () => {
  const component = renderer.create(
    <VisitorForm />,
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
  
});