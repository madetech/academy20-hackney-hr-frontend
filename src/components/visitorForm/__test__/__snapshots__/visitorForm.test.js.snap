// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visitor form is rendered 1`] = `
<div>
  <form
    data-testid="visitor-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="fname"
    >
      Your first name:
    </label>
    <br />
    <input
      id="fname"
      name="fname"
      type="text"
    />
    <br />
    <label
      htmlFor="lname"
    >
      Your last name:
    </label>
    <br />
    <input
      id="lname"
      name="lname"
      type="text"
    />
    <br />
    <label
      htmlFor="cnumber"
    >
      Your contact number:
    </label>
    <br />
    <input
      id="cnumber"
      name="cnumber"
      type="text"
    />
    <br />
    <label
      htmlFor="ename"
    >
      The employee you are
      <br />
      visiting:
    </label>
    <br />
    <input
      id="ename"
      name="ename"
      type="text"
    />
    <br />
    <label
      htmlFor="time"
    >
      Time of arrival (24hr):
    </label>
    <br />
    <input
      className="input-time"
      id="time"
      name="time"
      type="time"
    />
    <br />
    <br />
    <input
      className="button-submit"
      type="submit"
      value="Submit"
    />
    <div>
      
    </div>
  </form>
</div>
`;
